File: src/my_project_typescript/easy_problems_typescript/from101to150/findFirstPalindromicStringInArray.ts
function firstPalindrome(words: string[]): string {
    let answer: string = ""

    function isPalindrome(word: string): boolean{
        let answer: boolean = true 
        let lenWord: number = word.length
        for (let i = 0; i < Math.floor(lenWord/2); i++){
            if (word[i] !== word[lenWord-1-i]){
                answer = false
                return answer
            }
        }
        return answer 
    }

    for (let word of words){
        if (isPalindrome(word)){
            answer = word
            return answer 
        }
    }

    return answer 

};

function isPalindrome(word: string): boolean{
    let answer: boolean = true 
    let lenWord: number = word.length
    for (let i = 0; i < Math.floor(lenWord/2); i++){
        if (word[i] !== word[lenWord-1-i]){
            answer = false
            return answer
        }
    }
    return answer 

}

console.log(isPalindrome("d"))

File: src/my_project/easy_problems/from201to250/find_first_palindromic_string_in_array.py
from typing import List 

class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        answer: str = ""
        def isPalindrome(word: str) -> bool:
            
            answer: bool = True
            len_word: int = len(word)
            for i in range(len_word//2):
                if word[i] != word[len_word-1-i]:
                    answer = False
                    return answer
            return answer 
        
        for word in words:
            if isPalindrome(word):
                answer = word
                return answer
        
        return answer
    
solution = Solution()

print(solution.firstPalindrome(words = ["abc","car","ada","racecar","cool"]))

print(solution.firstPalindrome(words = ["notapalindrome","racecar"]))

print(solution.firstPalindrome(words = ["def","ghi"]))

    
